After login doesn't redirect to home

KL13NT opened the pull request:
# Issue #1 

## Expected
Login component is supposed to re-direct to the home component upon login and when the token is deleted manually the user should be able to access login again. 

## Actual
Login didn't redirect properly and deleting the token manually didn't change the app state

This PR fixes the issues by using a custom `AuthRoute` that makes sure the user is authenticated before navigation to /home. Login and Signup have also been updated to check whether the user is logged in upon mounting.

baymac commented on client/src/pages/login.js:
```diff
@@ -64,12 +71,11 @@ class login extends Component {
           loading: false
         });
-        this.props.history.push("/"); // Go to home page
+        this.props.history.push('/')
```
I am using `prettier - code formatter` vs code extension. With `Ctrl+shift+I` (windows and linux) it formats the code. I request you kindly reformatted all files with the extension so that we maintain uniformity. My suggestion is this if you want to keep contributing to this repository. Otherwise I will later find time to push these changes to the pr branch.

KL13NT replied:
I'm not seeing a config file for Prettier, can I safely assume you're working with the defaults?

baymac replied:
Yeah, I have all defaults.

baymac replied:
In case you want to add some configs feel free to do so, we can discuss what's best.

baymac commented on client/src/pages/signUp.js:
```diff
@@ -73,7 +82,7 @@ class signUp extends Component {
           loading: false
         });
-        this.props.history.push("/"); // Go to home page
+        this.props.history.push("/login"); // Go to home page
```
Remove comment.

baymac commented on client/src/util/utils.js:
```diff
@@ -0,0 +1,18 @@
+  if(authenticated === true) setAuthenticated(false)
+  return false
+
+}
```
Need a newline feed. It's a good practice. You can also enable in your editor.

baymac commented on client/src/App.js:
```diff
@@ -4,56 +4,58 @@ import "./App.css";
 
 
 // Pages
 import home from "./pages/home";
```
Wondering why you didn't capitalize the home component?

KL13NT replied:
I totally missed that, capitalised it in the latest commit

KL13NT commented:
Okay, so I formatted all my modifications using the default Prettier settings and removed unnecessary comments and console logs in a different branch. Do I PR to main from that branch now?

baymac commented:
I think you can push the changes the to this pr branch only. Let's try to keep all changes requested and changes in single pr only unless stated.

KL13NT commented:
Well, you had requested a different branch other than master. I formatted the code using Prettier and removed the extra comments and console logs. Also renamed `Home.js`.

baymac commented:
Yes, LGTM. But I will wait some time before merging this into master. I am trying to implement redux to see how it works and then we can compare and discuss about it then merge this. Need 3-4 hours more. Are you in UTC+2 timezone?

KL13NT commented:
I am, yes. If the app is planned to be so big that most of it is gonna rely on centralised state management, then absolutely go for Redux. If you're implementing Redux for the authentication part only, then I'd say it's a bit of an overkill.

baymac commented:
Thanks for the suggestion. I think we will need it so I'll implement redux
but keeping the PR shouldn't be an issue.

On Wed 2 Oct, 2019, 17:43 Nabil Tharwat, <notifications@github.com> wrote:

> I am, yes. If the app is planned to be so big that most of it is gonna
> rely on centralised state management, then absolutely go for Redux. If
> you're implementing Redux for the authentication part only, then I'd say
> it's a bit of an overkill.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/baymac/client-reader-auth/pull/2?email_source=notifications&email_token=AFQCTMA2BFST4IQNTGJFA7TQMSF5XA5CNFSM4I4USC32YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAEQ3LY#issuecomment-537464239>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFQCTMAL4FYA46K7EO3J3ZTQMSF5XANCNFSM4I4USC3Q>
> .
>
